src/main/java/com/shapesecurity/shift/es2017/codegen/WebSafeCodeGen.java

```diff
@@ -67,7 +68,9 @@ public CodeRep reduceLiteralRegExpExpression(@Nonnull LiteralRegExpExpression no
 		if (!safened.equals(node.pattern + "/")) {
 			throw new RuntimeException("WebSafeCodegen cannot safely output ASTs containing regex literals which require escaping");
 		}
-		return factory.token("/" + safened + buildFlags(node));
+		// see https://html.spec.whatwg.org/multipage/scripting.html#restrictions-for-contents-of-script-elements
+		String maybeLeadingSpace = startsWithScript.test(node.pattern) ? " " : "";
```

Comment:
I think we should be testing `safened`. But I'm also confused why that includes the trailing `/`. Don't we need to guarantee that `safe` is not replacing the trailing `/` in any way?

Reply:
Regardless, we should ensure that `safened` has not become an invalid RegExp.

Reply:
We don't use `safened` if any changes are made at all, preferring instead to throw. So I don't know what "become" means in this context.